compiler-rt/lib/asan/asan_descriptions.cpp

```diff
@@ -444,6 +444,15 @@ AddressDescription::AddressDescription(uptr addr, uptr access_size,
     data.kind = kAddressKindShadow;
     return;
   }
+
+  // Check global first. On AIX, some global data defined in shared libraries
+  // are put to the STACK region for unknown reasons. Check global first can
+  // workaround this issue.
+  if (GetGlobalAddressInformation(addr, access_size, &data.global)) {
+    data.kind = kAddressKindGlobal;
+    return;
+  }
+
```

Comment:
@jakeegan, the comment makes it sound like the problem being solved is not well-understood.
For further analysis, can we verify what fails when this change is not present?
I think we may want to leave a TODO comment here in case there is a different solution.